### ISSUE TITLE
Attributes from the previous paragraph element are copied to new paragraph when pressing return key

### ISSUE THREAD

**Issue by stevehanson** [yabwe/medium-editor#764]
I've noticed that if I have custom attributes on a paragraph tag and I create a new element after it, the attributes are copied over to the new element. Any thoughts on how I could get the editor to just create a blank `<p>` on new paragraphs? This issue isn't happening with headings.

![medium](https://cloud.githubusercontent.com/assets/1691324/9146604/a9faf082-3d23-11e5-9d6d-a37f9622285d.gif)

**Comment by nmielnik** [comment]
I was able to see this in both Chrome and Firefox, I'm not sure if it happens in other browsers.

So from what I can tell, it looks like the browser is cloning the attributes on the element each time to you hit enter.  I'm guessing this is useful so that if you have some special styling on a `<p>` (class, style, etc.) that hitting enter will make the next paragraph have the same styles.

Since the browser is inserting this, there are really 2 options:
1. Detect cases where this duplicated `<p>` would be generated (maybe just whenever the user hits <kbd>ENTER</kbd>?) and insert your own `<p>`.  You would still want to duplicate some attributes like `style` and `class` but then you could ensure nothing else is cloned.
1. Detect cases where this happens, and find the `<p>` that was generated by the browser and strip off all attributes other than the ones you want to keep cloned.

Generally, it has been easier to do something like 2, as opposed to something like 1, but I think those would be some of your options for now.

**Comment by stevehanson** [comment]
Thanks for the direction, @nmielnik. I'll give the options you recommended a go.

**Comment by wallylawless** [comment]
@stevehanson did you end up having any luck with this? I'm looking to do something similar.

**Comment by stevehanson** [comment]
@wallylawless no, I ended up just changing how I was doing things so that I didn't need to use the attributes on the top-level tags. I didn't spend too much more time trying to get it to work though, so that doesn't mean @nmielnik's suggestions wouldn't have worked.

**Comment by nmielnik** [comment]
From @nicholasalanbrown in #842 (duplicate)

> This is a terrific library with a very extensive readme, but I'm struggling to figure out how I can have the editor create paragraph IDs any time the user creates a new paragraph. I tried handling this after the submit action by replacing <p> tags with unique IDs a la <p id="1234"> and that works fine, but if I edit the markup again with the editor and create a new paragraph it simply copies down the unique ID from above.

> I'm trying to get something like SideComments working and this is a key part. Any help would be much appreciated!

**Comment by nicholasalanbrown** [comment]
Thanks for linking @mmielnik. Unfortunately I do need the browser to only create blank tags and I'm not quite sure how to implement either approach above...

**Comment by nmielnik** [comment]
@nicholasalanbrown there doesn't appear to be a simple way to automatically add attributes to newly created paragraphs, nor does there appear to be a simple way to prevent the `contenteditable` from automatically duplicating attributes from one `<p>` to a newly created `<p>`.

I recommended 2 options to attempt to handle this via changes to medium-editor early in this issue. For both, you might try attaching to when the user presses <kbd>enter</kbd> and do some cleanup work. 

You could also consider detecting when the user presses <kbd>enter</kbd> and instead of allowing the browser to create a new `<p>`, instead create a new `<div>` and a new instance of MediumEditor.  This way, you can control what attributes are on the elements and not worry about the browser removing/copying them.
